clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1023,6 +1135,7 @@ class SyclKernelBodyCreator
   InitializedEntity VarEntity;
   CXXRecordDecl *KernelObj;
   llvm::SmallVector<Expr *, 16> MemberExprBases;
+  llvm::SmallVector<size_t, 16> ArrayIndices;
```

Comment:
Can we not handle this with the offset calculations we've been doing?

Reply:
No. The current offset calculations deal with nested structs only. In a case such as Accessor a[2][2][2] for example, the struct offset code does not come into play at all. The offset calculations for arrays need to be done alongside the offset calculations for structs, and interleaved with each other for cases such as "struct { int i; Accessor a [2]; } d[3][3][3];".

Reply:
Right, but why can't we use the enter/leave array functions to modify where we are in the offset to figure out how to put this in the array?  Storing a list of array indices seems like the wrong way about this.